apps/image-server/server.js

```diff
@@ -175,6 +191,141 @@ app.post('/images',
     });
 
 
+
+const documentMulterConfig = {
+  onError: function (err, next) {
+    console.error(err);
+    next(err);
+  },
+  fileFilter: function (req, file, cb) {
+    const allowedTypes = [
+      'application/pdf',
+      'application/msword',
+      'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+      'application/vnd.ms-excel',
+      'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+      'application/vnd.ms-powerpoint',
+      'application/vnd.openxmlformats-officedocument.presentationml.presentation'
+    ];
+
+    if (allowedTypes.indexOf(file.mimetype) === -1) {
+      req.fileValidationError = 'goes wrong on the mimetype';
+      return cb(null, false, new Error('goes wrong on the mimetype'));
+    }
+
+    const forbiddenChars = /[\\/:]/;
+    if (forbiddenChars.test(file.originalname)) {
+      req.fileValidationError = 'Forbidden characters in filename';
+      return cb(null, false, new Error('Forbidden characters in filename'));
+    }
+
+    cb(null, true);
+  },
+  storage: multer.diskStorage({
+    destination: function (req, file, cb) {
+      cb(null, process.env.DOCUMENTS_DIR || 'documents/');
+    },
+    filename: function (req, file, cb) {
+
+      const uniqueFileName = createFilename(file.originalname)
+
+      cb(null, uniqueFileName);
+    }
+  })
+}
+
+documentMulterConfig.dest = process.env.DOCUMENTS_DIR || 'documents/';
+const documentUpload = multer(documentMulterConfig);
+
+app.get('/document/*',
+  function (req, res, next) {
+    req.url = req.url.replace('/document', '');
+
+    /**
+     * Pass request en response to the imageserver
+     */
+    // return res.download(`${process.env.APP_URL}/document/${req.url}`);
+    return res.download(`documents/${req.url}`);
+  });
+
+const allowedExtensions = [
+  'pdf',
+  'doc',
+  'docx',
+  'xls',
+  'xlsx',
+  'ppt',
+  'pptx'
+];
```

Comment:
Waarom is hier een check op de extension als we hierboven in de documentMulterConfig al een mimetype filter hebben?